exercises.py

```diff
@@ -3,26 +3,44 @@
 """
 
 def reverse(s: str) -> str:
-    """
-    Reverse string using recursive approach
-    """
+    if len(s) <= 1:
+	    return s
+	else:
+	    return reverse(s[1:])  + s[0]
 
 def fib_rec(n: int) -> int:
-    """
-    Recursive fibonacci
-    """
+    if n == 0 or n == 1:
+        return 1
+    else:
+        return fib_rec(n - 1) + fib_rec(n - 2)
 
+cache = {}
 def fib_dyn(n: int) -> int:
-    """
-    Fibonacci using dynamic programming (w/ a cache)
-    """
+     if n == 0 or n == 1:
+        return 1
+    if n in cache:
+        return cache[n]
+    else:
+        cache[num] = fib_dyn(n - 1) + fib_dyn(n - 2) 
+        return cache[num]
+
 
 def fib_iter(n: int) -> int:
-    """
-    Iterative approach to solving the fibonacci problem
-    """
+    last = 1
+    two_before = 1
+    if n == 0 or n == 1:
+        return 1
+    for i in range(2, n + 1):
+        n = last + two_before
+        two_before = last
+        last = n
+    return n
 
 def to_str(n: int, base: int) -> int:
```

Comment:
Run tests to catch errors like with the type hint here. What data type are you returning?